soa-derive-internal/src/refs.rs

```diff
@@ -142,6 +142,18 @@ pub fn derive(input: &Input) -> TokenStream {
             }
         }
 
+        impl<'a>  From<#ref_name<'a>> for #name where #( for<'b> #fields_types: Clone, )* {
+            fn from(value: #ref_name<'a>) -> #name {
+                value.to_owned()
+            }
+        }
+
+        impl<'a>  From<&'a #ref_name<'a>> for #name where #( for<'b> #fields_types: Clone, )* {
+            fn from(value: &'a #ref_name<'a>) -> #name {
+                value.to_owned()
+            }
+        }
+
```

Comment:
Not sure if I like having `.into` and `From::from` with the same meaning as clone/to_owned. Why is this required?

Reply:
This is for parity with `Vec<T>`: https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-From%3C%26%5BT%5D%3E-for-Vec%3CT%3E

`to_owned` is an inherent method on the generated type, but its existence on `&[T]` is actually part of the `ToOwned` trait: https://doc.rust-lang.org/std/primitive.slice.html#impl-ToOwned-for-%5BT%5D, and since I can't implement `ToOwned`/`Borrow` for SoA types, this was a convenient work-around as well.

Reply:
The argument for this is that `to_owned` is an inherent method, pretending to be an implementation of `ToOwned` via the blanket implementation: https://doc.rust-lang.org/std/borrow/trait.ToOwned.html#impl-ToOwned-for-T. 

Since `ToOwned` depends upon `Borrow`, which can't be implemented for SoA types, I needed some "type" level way of expressing this.

Reply:
I don't understand what you mean here. You want to implement `ToOwned` but can't? The point I don't understand is using `From` for this.

Reply:
ok, this makes sense. Thanks for the detailed explanation!